.storybook/utils/docscript.js

```diff
@@ -1,70 +1,236 @@
+ *              "type": string,
+ *              "description"
+ *          }, ...]
+ *     }, ...
+ *  },
+ *  "events": {
+ *      "event1": {
+ *          "description": string,
+ *          "params": [{
+ *              "name": string,
+ *              "type": string,
+ *              "description"
+ *          }, ...]
+ *     }, ...
+ * }
+ */
+export function getDocsForClass(cls) {
+    // if data already in cache return it
+    if (cache[cls]) {
+        return cache[cls];
+    }
+
+    // prepare result
+    var entry = {
+        description: '',
+        properties: {},
+        methods: {},
+        events: [],
+        parentClasses: [],
+        interfaces: []
+    };
+
+    // cache it
+    cache[cls] = entry;
+
+    // get docs for class
+    var relevantDocs = docs.filter(doc => doc.name === cls || doc.memberof === cls);
+    relevantDocs.forEach(doc => {
+        if (doc.undocumented) return;
+
+        if (doc.classdesc) {
+            entry.description = doc.classdesc;
+        }
+
+        if (doc.augments) {
+            entry.parentClasses = entry.parentClasses.concat(doc.augments);
+        }
+
+        if (doc.mixes) {
+            entry.interfaces = entry.interfaces.concat(doc.mixes);
+        }
+
+        if (doc.properties) {
+            doc.properties.forEach(prop => {
+                if (prop.undocumented || !prop.description) return;
+
+                var name = prop.name.replace('args.', '');
+                entry.properties[name] = {
+                    description: prop.description,
+                    type: prop.type.names[0],
+                    defaultValue: prop.defaultvalue
+                };
+            });
+        }
+
+        if ((doc.kind === 'class' || doc.kind === 'member') && doc.description) {
+            var data = {
+                name: doc.kind === 'class' ? 'constructor': doc.name,
+                description: doc.description,
+            };
+            if (doc.params) {
+                data.params = doc.params.map(param => {
+                    return {
+                        name: param.name,
+                        type: param.type.names[0],
+                        description: param.description
+                    };
+                });
+            }
+
+            entry.methods[data.name] = data;
+        }
+
+        if (doc.kind === 'event' && doc.description) {
+            var data = {
+                name: doc.name,
+                description: doc.description,
+            };
+            if (doc.params) {
+                data.params = doc.params.map(param => {
+                    return {
+                        name: param.name,
+                        type: param.type.names[0],
+                        description: param.description
+                    };
+                });
+            }
+
+            entry.events[data.name] = data;
+        }
+    });
+
+    return entry;
 }
 
+export function getStoryBookDocs(data) {
```

Comment:
Can this be renamed getStorybookDocs to match the official name?

Reply:
Done